### DISCUSSION TITLE
编译好的boringSSL没发现生成ssl.h头文件在哪#include "openssl/ssl.h"

### DISCUSSION THREAD

**Discussion by tickmoo** [Tencent/tquic#315] [Q&A (问题和回答)]
使用windwos平台编译boringssl，生成了libssl.a和libcrypto.a，并且build.rs构建println!("cargo:rustc-link-lib=static=ssl");
    println!("cargo:rustc-link-lib=static=crypto");也成功打进去了，但是tquic.h里面需要的openssl/ssl.h头文件在哪里找？

**Comment by glime-ai** [comment]
在Windows平台上编译BoringSSL并生成了`libssl.a`和`libcrypto.a`之后，`tquic.h`文件中需要的`openssl/ssl.h`头文件可以在BoringSSL的源代码目录中找到。具体步骤如下：

1. **下载BoringSSL源码**：确保你已经下载了BoringSSL的源码，可以通过以下命令获取：
    ```sh
    git clone https://boringssl.googlesource.com/boringssl
    ```

2. **查找头文件**：在BoringSSL的源码目录中，头文件通常位于`include`目录下。你可以在`boringssl/include/openssl/`目录中找到`ssl.h`文件。

3. **配置头文件路径**：在你的构建脚本或项目配置中，确保包含BoringSSL的头文件路径。例如，在`build.rs`中添加：
    ```rust
    println!("cargo:include=path/to/boringssl/include");
    ```

这样，编译器在编译时就能找到`openssl/ssl.h`头文件。

This comment was generated by [Glime](https://glimelab.ai/?utm_source=44520fce49ba43798485a628d6928e22&utm_medium=github).

**Comment by iyangsj** [comment]
@tickmoo 目前暂不支持 Windows平台

参见 https://tquic.net/zh/docs/getting_started/installation/#%E6%94%AF%E6%8C%81%E5%B9%B3%E5%8F%B0

**Comment by yflisa** [comment]
@iyangsj 现在文档 （https://tquic.net/zh/docs/getting_started/installation/#%E6%94%AF%E6%8C%81%E5%B9%B3%E5%8F%B0）中支持windows 编译，tquic 依赖的也是 boringssl 吗？还是直接换成 openssl 的库也可以？

**Reply by xiaofei0800** [reply]
tquic依赖boringssl，通过git submodule update --init --recursive拉取依赖。不能替换openssl，两者接口和功能有较大差异。
